be/src/storage/vectorized/delta_writer.cpp

```diff
@@ -172,36 +182,57 @@ Status DeltaWriter::_init() {
 
     _tablet_schema = writer_context.tablet_schema;
     _reset_mem_table();
-
-    // create flush handler
     _flush_token = _storage_engine->memtable_flush_executor()->create_flush_token();
-    _is_init = true;
+    _set_state(kWriting);
     return Status::OK();
 }
 
-Status DeltaWriter::write(Chunk* chunk, const uint32_t* indexes, uint32_t from, uint32_t size) {
-    SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(_mem_tracker.get());
-
-    if (_is_cancelled) {
-        return Status::OK();
-    }
-    if (!_is_init) {
-        RETURN_IF_ERROR(_init());
-    }
-
-    bool flush = false;
-    try {
-        flush = _mem_table->insert(*chunk, indexes, from, size);
-    } catch (std::bad_alloc const&) {
-        return Status::MemoryLimitExceeded("Mem usage has exceed the limit of BE");
+Status DeltaWriter::write(const Chunk& chunk, const uint32_t* indexes, uint32_t from, uint32_t size) {
+    SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(_mem_tracker);
+    Status st;
+    auto state = _get_state();
+    switch (state) {
+    case kUninitialized:
+    case kCommitted:
+    case kAborted:
+    case kClosed:
+        return Status::InternalError(fmt::format("cannot write delta writer in {} state", _state_name(state)));
+    case kWriting:
+        bool full = _mem_table->insert(chunk, indexes, from, size);
```

Comment:
If the memory is enough, there is not necessary to check the mem_limit of one load.
The purpose is to control the total memory of all load jobs instead of only one load job.

Reply:
A user can still specify the memory limit for a single load job now, and the current implementation is compatible with the previous implementation. Note that if a user does not specify a memory limit, the per-job memory limit is equal to the total memory limit.